### DISCUSSION TITLE
Quais são as leituras obrigatórias do momento considerando desenvolvedoras frontend?

### DISCUSSION THREAD

**Discussion by dayanyrec** [frontendbr/forum#1034] [Discussão]
Na issue #475 @RatoX comentou

> Só lembrando que qualquer código sem teste é código legado, Working Effectively with Legacy Code Leitura obrigatória para quem tá virando senior ou já é um(a)!

Aproveitando o comentário para discutir quais as leituras obrigatórias do momento considerando desenvolvedoras frontend até a senioridade?

**Comment by dayanyrec** [comment]
Quando digo considerando desenvolvedoras frontend é para dar **foco** em **front** e incluir principalmente leituras como [Eloquent Javascript](http://eloquentjavascript.net/index.html) e não só leituras genéricas como [TDD. Desenvolvimento Guiado por Testes](https://www.amazon.com.br/TDD-Desenvolvimento-Guiado-por-Testes/dp/857780724X/).

Enfim, vejo a lista nascendo:

- [Eloquent Javascript](http://eloquentjavascript.net/index.html)
- [TDD. Desenvolvimento Guiado por Testes](https://www.amazon.com.br/TDD-Desenvolvimento-Guiado-por-Testes/dp/857780724X/)
- [Working Effectively with Legacy](https://www.amazon.com.br/Working-Effectively-Legacy-Michael-Feathers/dp/0131177052)

**Comment by annacruz** [comment]
Apesar de front não ser a minha especialidade, eu poderia recomendar a versão nova do Refactoring do Martin Fowler [1] que vai sair. O livro original tinha exemplos em Java, mas a versão nova vem com os exemplos em javascript. Acredito que é uma ótima pedida.

[1] [https://martinfowler.com/articles/201803-refactoring-2nd-ed.html]()

**Comment by laurindo** [comment]
Esses livros são muito bons. Ontem estava lendo o livro da casa do código *"introdução a arquitetura e design de sistemas". Considero uma leitura essencial.

*Não coloquei o link do livro pois estou no celular.

**Comment by ninetails** [comment]
Complementando as listas anteriores, no caso de JavaScript:

* [JavaScript: the right way](http://jstherightway.org/pt-br/)

Fora de Frontend, além do Refactoring:
* **Padrões de Arquitetura de Aplicações Corporativas**, também do Martin Fowler
* **Clean Code(r)** da vida (repito: acho hype mas nada mais é que um compilado/resumo de bons livros que deveriam ser lidos separada e integralmente)

**Comment by ghost** [comment]
Oi @dayanyrec , tudo bem contigo? Bom vou tentar te ajudar com o que adiquiri de bagagem com o tempo: 

- Falando de linguagem eu consideraria esses:

[The good parts](https://www.amazon.com.br/JavaScript-Good-Parts-Douglas-Crockford/dp/0596517742)

[Secrets of the JavaScript Ninja](https://www.amazon.com/Secrets-JavaScript-Ninja-John-Resig/dp/193398869X/ref=asap_bc?ie=UTF8)

[You Don't Know JS series](https://www.amazon.com/You-Dont-Know-Js-Book/dp/B01AY9P0P6)

Além de EloquentJS já citado.

- Uma menção honrosa de algumas pessoas que produzem conteúdos fantástico:

[Axel Rauschmayer](http://dr-axel.de/)

[Eric Elliot](https://ericelliottjs.com/)

Também vale ficar de olho no conteúdo do [Frontend Masters](https://frontendmasters.com) e do [Egghead](https://egghead.io)

Ao beber dessas fontes já te darão boa formação e te moldará para uma "senioridade" com bastante senso critico e eloquência para discutir qualquer assunto do meio (frontend).

ps: Lembrando que "senioridade" é tempo, isso quer dizer que tudo aquilo que muitos defendem aqui sobre aprender bem a base ainda é o mais valioso!

**Comment by luizbills** [comment]
Minha recomendação: [7 Rules for Creating Gorgeous UI](https://medium.com/@erikdkennedy/7-rules-for-creating-gorgeous-ui-part-1-559d4e805cda)

**Comment by lucasfcosta** [comment]
IMO chamar de leituras "obrigatórias" é meio exagerado se não for especificado um tópico, afinal cada um tem seu foco de estudos, então recomendo sempre que a galera postar uma recomendação adicionar também uma keyword ou uma tagline pra especificar do que se trata e ficar mais fácil separar depois.

Anyway, aqui vai uma lista dos meus livros preferidos:

* CompSci General
    * The algorithm design manual (se tem algo que é obrigatório é isso aqui)
* Matemática, Lógica Combinatória e Functional Programming (inclua aí computabilidade, por motivos de: tudo é muito interligado)
    * To Mock a Mockingbird (provavelmente o mais fascinante dessa lista)
    * Learn You Haskell for Great Good
    * The little schemer, The seasoned schemer
    * Functional Data Structures
    * Structure and Interpretation of Computer Programs
    * Papers:
        * [The Original Lambda Papers (MARAVILHOSO)](http://library.readscheme.org/page1.html)
        * Computability and Recursion - Robert Soare
        * The History and Concept of Computability - Robert Soare
        * Computability and Incomputability - Robert Soare
        * A Tutorial Introduction to Lambda Calculus - Raul Rojas
        * Lamda Calculli with Types - Henk Barendregt
        * Proofs and Types (Lambda Calculus) - Jean Yves Girard
        * The Impact of Lambda Calculus in Logic and Computer Science - Henk Barendregt
        * Type Theory and Functional Programming - Simon Thompson
* Architecture (não gosto muito do termo), DevOps, Proccess (tudo na mesma categoria por ser bem interligado também)
    * Continuous Delivery (do Jez Humble e David Farley, excelente livro)
    * The Mythical Man Month
    * Building Microservices - Sam Newman
    * Papers:
        * No Silver Bullet - FP Brooks
* JavaScript
    * Eloquent JavaScript (para iniciantes)
    * JavaScript The Good Parts
    * MDN (não é um livro, mas são os melhores docs ever)

Como acho que deu pra notar eu sou pretty heavy em FP/Math/Fundamentals do que em frameworks/tecnologias em específico porque eu acho que isso é que te ajuda a desenvolver seu raciocínio e consequentemente entender melhor qualquer outro tópico.

PS: Continuem postando recomendações, já to anotando ❤️ 

Cheers,
Lucas

**Comment by felipefialho** [comment]
Linkando: #229 #253

**Comment by vinicius-batista** [comment]
Eu gostei muito de ler a serie [Composing Software](https://medium.com/javascript-scene/composing-software-an-introduction-27b72500d6ea) do Eric Elliot e todos seus exemplos utilizam javascript
